### PR TITLE
Add option to wait for signald.send() confirmation

### LINKED ISSUE #28: Add option to wait for signald.send() confirmation
Currently, semaphore just sends a message to Signald's socket and as client there is no chance to see whether sending was successful. But Signald answers with a [SendResponse](https://docs.signald.org/structures/v1/SendResponse.html), which contains whether sending a message was successful or not.

Imho, we should add an option like `confirm=False` which should be False by default - because this is the current behavior - but if True it listens for a `SendResponse`  and returns whether sending was successful or not. What do you think?

### PR DESCRIPTION
**Pull request by eknoes**
Resolves #28 

I use 2 different sockets for now, so we do not interfere with receiving messages if waiting for a response to our send message. This PR makes `send_message -> bool` and logs warnings if a message could not be sent.

We could also add this for `reply_message` or do you think this is not really necessary?

### CODE REVIEW

File: semaphore/message_sender.py [hunk 1]
```diff
@@ -61,6 +64,37 @@ async def send_message(self, receiver, body, attachments=None):
...
+            response = response_wrapper['data']
+            results = response.get("results")
+            if not results:
+                continue
+
+            for result in results:
+                if result['address'].get('uuid') == receiver or result['address'].get('number') == receiver:
+                    if result.get('success'):
+                        return True
+                    return False
```
Comment (eknoes) [inline semaphore/message_sender.py:94]: I am not really confident with Pythons async behaviour, but could this here lead to a race condition, when 2 messages are sent but the second one is confirmed earlier by signald?

Comment (lwesterhof) [reply semaphore/message_sender.py:98]: Yeah, looks like this could lead to trouble.

**Review by lwesterhof — CHANGES_REQUESTED** [review]
It looks like you are not using the id field of signald (https://gitlab.com/signald/signald/-/issues/84). Is that correct?

**Comment by eknoes** [comment]
I was not aware of the id field before Finns comment on the issue, I'll add that as soon as I have the time for that! But that would still not solve the problem of getting a receipt for a second message before the first. I could just add an array of confirmed IDs which is checked before the socket is read. But I have no idea of how safe it is in terms of "thread safety" of the async framework. You have any idea? Otherwise I would try to read a bit about those issues in Python before implementing it, but that will take some time as I am quite busy for the next weeks.

File: semaphore/message_sender.py [hunk 2]
```diff
@@ -59,8 +65,44 @@ async def send_message(self, receiver, body, attachments=None):
         if attachments:
             bot_message["attachments"] = attachments
 
+        self.signald_message_id += 1
+        bot_message['id'] = self.signald_message_id
```
Comment (eknoes) [inline semaphore/message_sender.py:69]: @lwesterhof Now I do use an incrementing value to identify the correct signald response as you requested

**Review by eknoes — COMMENTED** [review]
@lwesterhof Now I do use an incrementing value to identify the correct signald response as you requested

**Comment by eknoes** [comment]
Sorry, I found some issues. I'll comment when I fixed them.

**Comment by eknoes** [comment]
I ran my tests on the wrong commit, but now I fixed the comparison, tested it and it works fine :)

**Comment by eknoes** [comment]
Is there any update or something I should do, to have this PR accepted?

**Comment by eknoes** [comment]
I just merged the current main branch to avoid a merge conflict.

**Review by lwesterhof — APPROVED** [review]

**Comment by eknoes** [comment]
Thanks for merging!

**Pull request merged**